File: stacks/go/src/adapters/clock.go
// Package adapters wraps I/O at the boundaries: HTTP, DB, queues, filesystem, time.
package adapters

import "time"

// Clock returns the current time. Replace with a fake in tests.
type Clock interface {
	Now() time.Time
}

type systemClock struct{}

func (systemClock) Now() time.Time { return time.Now().UTC() }

// SystemClock is the default Clock backed by time.Now in UTC.
var SystemClock Clock = systemClock{}

File: stacks/python/src/adapters/clock.py
from __future__ import annotations

from dataclasses import dataclass
from datetime import UTC, datetime
from typing import Protocol


class Clock(Protocol):
    def now(self) -> datetime: ...


@dataclass(frozen=True)
class _SystemClock:
    def now(self) -> datetime:
        return datetime.now(tz=UTC)


system_clock: Clock = _SystemClock()
